exo-go/src/process_helpers/process.go

```diff
@@ -119,47 +129,41 @@ func (p *Process) Wait() error {
 	return p.Cmd.Wait()
 }
 
-func (p *Process) waitFor(condition func(string) bool) error {
-	var output string
-	for {
-		text, err := p.readStdoutPipe()
-		if err != nil {
-			return err
-		}
-		output = output + text
-		if condition(output) {
-			return nil
-		}
+func (p *Process) waitFor(condition func(string) bool, err chan<- error) {
+	p.outputMutex.Lock()
+	if condition(p.Output) {
+		err <- nil
+	} else {
+		id := uuid.NewV4().String()
+		p.AddOutputFunc(id, func(output string) {
+			if condition(output) {
+				err <- nil
+				p.RemoveOutputFunc(id)
+			}
+		})
 	}
+	p.outputMutex.Unlock()
```

Comment:
Lock/unlock the output to prevent it from being updated while this function is checking against the pre-existing output and adding a new callback if needed